ConfigSpace/io/pcs_new.py

```diff
@@ -0,0 +1,452 @@
+            child = configuration_space.get_hyperparameter(child_name)
+            if '(' not in condition:
+                parent_name = condition[2]
+                parent = configuration_space.get_hyperparameter(parent_name)
+                operation = condition[3]
+                # in template must be treated differently from the rest
+                if operation == 'in':
+                    restrictions = condition[5:-1:2]
+                    if len(restrictions) == 1:
+                        condition = EqualsCondition(child, parent, restrictions[0])
+                    else:
+                        condition = InCondition(child, parent, values=restrictions)
+                    condition_objects.append(condition)
+                
+                else:
+                    restrictions = condition[4]
+                
+                    if operation == '==':
+                        condition = EqualsCondition(child, parent, restrictions)
+                    elif operation == '!=':
+                        condition = NotEqualsCondition(child, parent, restrictions)
+                    elif operation == '<':
+                        condition = LessThanCondition(child, parent, restrictions)
+                    elif operation == '>':
+                        condition = GreaterThanCondition(child, parent, restrictions)
+            
+                    condition_objects.append(condition)
+            else:
+                parent_name = condition[1]
+                parent = configuration_space.get_hyperparameter(parent_name)
+                operation = condition[2]
+                # in template must be treated differently from the rest
+                if operation == 'in':
+                    restrictions = condition[4:-1:2]
+                    if len(restrictions) == 1:
+                        condition = EqualsCondition(child, parent, restrictions[0])
+                    else:
+                        condition = InCondition(child, parent, values=restrictions)
+                    condition_objects.append(condition)
+            
+                else:
+                    restrictions = condition[3]
+            
+                    if operation == '==':
+                        condition = EqualsCondition(child, parent, restrictions)
+                    elif operation == '!=':
+                        condition = NotEqualsCondition(child, parent, restrictions)
+                    elif operation == '<':
+                        condition = LessThanCondition(child, parent, restrictions)
+                    elif operation == '>':
+                        condition = GreaterThanCondition(child, parent, restrictions)
+        
+                    condition_objects.append(condition)       
+        if len(condition_objects) > 1:
+            for connective in connectives:
+                if connective == '&&':
+                    and_conjunction = AndConjunction(*condition_objects)
```

Comment:
I don't think this is correct. This statement adds all conditions with the same boolean operator to the configuration space. Instead of doing this, you need to first add all adjacent `||` to an `OrConjunction` and then connect all left `&&` with a single `AndConjunction`.